### DISCUSSION TITLE
[Feature] Handle existing XMP sidecars without file extension

### DISCUSSION THREAD

**Discussion by patrickli** [immich-app/immich#6537] [Feature Request]
### The feature

Immich can read existing sidecars in the format of `{originalFilename}.{originalFileExtension}.xmp`. However software such as Lightroom writes the xmp as `{originalFilename}.xmp`, without the extension (jpg, etc).

There are a few comments in the code that mention this, for example
* https://github.com/immich-app/immich/blob/main/cli/src/cores/models/asset.ts#L37
* https://github.com/immich-app/immich/blob/main/server/src/domain/library/library.service.ts#L240

I think the best way is to handle both cases, ie, find sidecars with and without the file extension.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by chrneu** [comment]
@alextran1502: Do you have an opinion on this topic? I would also appreciate if this could be handled both ways. 
https://immich.app/docs/features/xmp-sidecars/

As a side remark, I just used a demo of photoprism. There you also have the possibility to upload the photo and the xmp directly together via web ui. Even though this is not a primary usecase, I think it could be a nice extension.

**Reply by alextran1502** [reply]
I think it is a good step forward to help support xmp file. So feel free to open a PR to put this into the project.

**Comment by kokulshan** [comment]
This support now exists in PR #7813
